### ISSUE TITLE
Authentik does not delete entities like users and groups from its DB when a user or a group were deleted from the external LDAP Catalog. 

### ISSUE THREAD

**Issue by headstack** [goauthentik/authentik#6408]
**Is your feature request related to a problem? Please describe.**
I delete a group on FreeIPA then click the SYNC button on Authentik's LDAP Federation. Sitting and waiting for Authentik is going to delete the group, but it isn't doing that.

I've got the following schema:
1. FreeIPA as an LDAP catalog contains users and groups
2. Authentik uses LDAP federation to sync users and groups from FreeIPA.

**Describe the solution you'd like**
Please, add functionality that could store all the entities, that Authentik is getting from the external LDAP catalog, up to date.

**Describe alternatives you've considered**
.

**Additional context**
Authentik version: 2023.6.1
Deployment type: K8S

**Comment by terion-name** [comment]
Yes, there should be an option to enable this

**Comment by phauch** [comment]
Even though there is nothing in the LDAP sync code to handle this, it is not impossible to implement this functionality. The LDAP sync already fetches every user it finds, and if the user already exists in Authentik, it gets updated. If the user does not exist, a new user is created in Authentik.

So, why not use this to compare the list of users fetched from LDAP against the list of users in Authentik? This way, users that exist only in Authentik (with the appropriate slug) and not in the LDAP site can be identified and deleted from Authentik.

Currently, we are evaluating how to address this scenario. With a significant number of users joining and leaving within a year, it's crucial to ensure that if a user is deleted, it is also removed from Authentik. Otherwise, the user would still have access to their resources, and we would not be GDPR compliant.

**Comment by pubyun** [comment]
we need a function to easy identify those users in Authentik that are not in LDAP source.

so we can delete those users easily.

it's good if  Authentik provide an switch which it can disable users which are not in LDAP source.

**Comment by tograss** [comment]
Would you like a pull request for this issue ? Or are you working internally on it ?

**Comment by rissson** [comment]
There are other things we need to rework in the ldap source before we can get to this, so not yet

**Comment by tograss** [comment]
Hi,
For those interested I have written a custom command which deactivates missing users and deletes users which are missing and have not login more than 6 month. 
As always I'm just some one on the internet. So make a backup, read the code and decide for your self. No guaranties for anything!
Get the script [ldap_custom_user_sync.txt](https://github.com/user-attachments/files/19630061/ldap_custom_user_sync.txt) and rename to .py
Then to use the script create a token in authentik and replace XXX in the script. 
Then use docker cp or mount the script via composer to  authentik/sources/ldap/management/commands/ 


Switch in the docker container and run the scrip with python manage.py slug_of_ldap_source
The following options are supported --dry-run do nothing just log, --no-delete only deactivate accounts do not delete them, --fix-pw 
if given fixes issue #11436  

I'm a bit unsure how users are associated with a source in authentik, the db layout suggests a source table but at least on my instance this is empty. So I used the path to identify which users belong to which ldap source. Again check if this is working for you.

**Comment by BeryJu** [comment]
This was added with 2025.6 https://docs.goauthentik.io/docs/releases/2025.6

Requires enabling **Delete Not Found Objects** on the LDAP source: https://docs.goauthentik.io/docs/users-sources/sources/protocols/ldap/

**Comment by dominic-r** [comment]
We're closing this in favor of this parent issue: #25258
